Applications/TextEditor/main.cpp

```diff
@@ -42,6 +43,16 @@ int main(int argc, char** argv)
         return 1;
     }
 
+    const char* preview_mode = "default";
+    const char* file_to_edit = nullptr;
+    Core::ArgsParser parser;
+    parser.add_option(preview_mode, "Preview mode, one of 'none', 'html', 'markdown', 'default'", "mode", 'm', "mode");
```

Comment:
I wouldn't call it `--mode`/`-m`. `--preview-mode` (with no short variant) sound better.

Reply:
why make it so darn hard to set a preview mode?
(no qualms about `--preview-mode` itself)

Reply:
Why hard? It's just that, in the context of a text editor, "mode" doesn't immediately mean "whether we're previewing Markdown or HTML" to me. "Preview mode" at least says a mode of *what* it is.

Reply:
nonono, I mean, why disable the short form?
`--preview-mode` makes more sense, that's fine by me.

Reply:
Because, IMO, neither `-p` nor `-m` are good shortenings of `--preview-mode`. `-m` would have been a good shortening of `--mode`. Something weird would work, like `-M`, for a system that doesn't support long options, but we don't have to do that.

That being said, I'm not inherently against it, just saying it doesn't make much sense to me. If you want it, I won't object.

Reply:
how about an (admittedly unorthodox) alias like "--pm"?
nevermind, terrible idea.

Reply:
I'll just remove the short version, some day we'll have tab-completions for these anyway.